Update PROFILES doc to list profile names

brian-avery opened the pull request:
This adds the name that should be used for installing a profile to the docs since this is what the installation docs point to. I couldn't find a name for the Kubernetes profile.



[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

brian-avery committed:
Update PROFILES.md

This adds the name that should be used for installing a profile to the docs since this is what the installation docs point to. I couldn't find a name for the Kubernetes profile.

istio-testing commented:
@brian-avery: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
release-notes_istio | 9e404a80f782378747b97fd46bd3d086b46e5234 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/release-notes_istio/1324067589462167552) | `/test release-notes_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@brian-avery: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
release-notes_istio | fa6e73ba0b4acdbb1e41bc0e2f63673a4460f75f | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/release-notes_istio/1324071486071246848) | `/test release-notes_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@brian-avery: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
release-notes_istio | f60e1c2d3f4ae827e25a8374ab5f6121448efd02 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/release-notes_istio/1324073475647737856) | `/test release-notes_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

brian-avery commented on manifests/profiles/PROFILES.md:
```diff
@@ -10,7 +10,7 @@ will provide further instructions. Additionally, please read the RFC requirement
+|Profile Name| K8s Platform         | Vendor        | Lead Maintainer | Lead Maintainer  |
+|------------| -------------------- | ------------- | --------------- | ---------------- |
+|openshift| OpenShift            | Red Hat / IBM | @jwendell       | @irisdingbj      |
+|| Upstream Kubernetes  | Istio         | Env WG          | discuss.istio.io |
```
Is discuss.istio.io actually the lead maintainer here?

jwendell replied:
In my opinion: Rename this file to vendor-profiles.md **or, better yet** Drop it entirely and put it in docs website somewhere.

If we decide to keep this file: Drop the upstream kubernetes file, as it is not a vendor.

brian-avery replied:
I like the idea of dropping this and adding it directly to the docs. Was surprised to see it here. 

I also ready this as there being a custom vendor profile specifically for Kubernetes but I was unable to find it hence my other question.

sdake replied:
line 16 is meant to represent the other profiles. The other profiles are meant to run on upstream K8s - and are maintained by the K8s community.

An improved way to represent this table is to show each profile and its maintianer.

I had hoped at some point `profile list` could parse this file and produce support information - hence why it is in repo. I don't see major harm in reproducing the docs in two places if istio.io is desired. I was simply trying to keep it simple 💯

brian-avery replied:
Thanks for the clarification @sdake. 


When you say "An improved way to represent this table is to show each profile and its maintianer." Are you referring to what I added in this PR? I think all of that info is already there. 

In my opinion, it would make the most sense if these were listed alongside the other profiles here: https://preliminary.istio.io/latest/docs/setup/additional-setup/config-profiles/. Right now we kind of say "here's our standard installation profiles. Look at the source code for more." And the second additional profile is just a conglomerate reference to the profiles you just came from. For now, it might make sense to have the duplicate content to make it clearer for users. 

Having the list auto-generated off of the profiles list sounds like it might be useful and we have other docs generated off of content from other repositories already, so potentially that content might be able to be used to update the istio.io page?

istio-testing commented:
@brian-avery: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 99d538b3e56eaff4d83da53fb05de10ea096c01f | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/unit-tests_istio/1324074034547134464) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

brian-avery commented:
/test unit-tests_istio

istio-testing commented:
@brian-avery: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 99d538b3e56eaff4d83da53fb05de10ea096c01f | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/unit-tests_istio/1324074034547134464) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 99d538b3e56eaff4d83da53fb05de10ea096c01f | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/integ-pilot-k8s-tests_istio/1324074034639409152) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

brian-avery commented:
/test integ-pilot-k8s-tests_istio

istio-testing commented:
@brian-avery: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 99d538b3e56eaff4d83da53fb05de10ea096c01f | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/integ-pilot-k8s-tests_istio/1324074034639409152) | `/test integ-pilot-k8s-tests_istio`
integ-security-k8s-tests_istio | 99d538b3e56eaff4d83da53fb05de10ea096c01f | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/integ-security-k8s-tests_istio/1324074034681352192) | `/test integ-security-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

jwendell commented on releasenotes/notes/addProfileNames.yaml:
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
```
I don't think a release note entry is necessary for this...

brian-avery replied:
I was hesitant at first, but it does change visible user behavior in that these are the instructions that users are directed to from Istio.io for installing profiles.

brian-avery replied:
Looking at our release notes docs though, release notes are required for:

    CLI changes
    API changes
    Configuration schema change
    Behavioral change
    Change in non-functional attributes such as efficiency or availability, availability of a new platform
    A warning about a deprecation
    Fix of a previous Known Issue
    Promoting features

This might be considered a CLI change, but I'd lean towards agreeing that there are no release notes needed.

sdake replied:
The release notes should have gone in with the PR that added the vendor profile. Since it didn't, I added it here instead. I think a release note is definitely needed, even if we decide as a community to store this information on istio.io (which makes sense to me).

brian-avery committed:
Delete addProfileNames.yaml

Release notes are required for: 


    CLI changes
    API changes
    Configuration schema change
    Behavioral change
    Change in non-functional attributes such as efficiency or availability, availability of a new platform
    A warning about a deprecation
    Fix of a previous Known Issue
    Promoting features

While this does affect the behavior for users, it doesn't fall into one of the categories above.

sdake commented on manifests/profiles/PROFILES.md:
```diff
@@ -10,7 +10,7 @@ will provide further instructions. Additionally, please read the RFC requirement
-| -------------------- | ------------- | --------------- | ---------------- |
-| OpenShift            | Red Hat / IBM | @jwendell       | @irisdingbj      |
-| Upstream Kubernetes  | Istio         | Env WG          | discuss.istio.io |
+|Profile Name| K8s Platform         | Vendor        | Lead Maintainer | Lead Maintainer  |
```
A space is needed between `|` or I believe the table will not render properly.

brian-avery replied:
This renders fine on GitHub. I can still add the space

sdake commented on manifests/profiles/PROFILES.md:
```diff
@@ -10,7 +10,7 @@ will provide further instructions. Additionally, please read the RFC requirement
-| OpenShift            | Red Hat / IBM | @jwendell       | @irisdingbj      |
-| Upstream Kubernetes  | Istio         | Env WG          | discuss.istio.io |
+|Profile Name| K8s Platform         | Vendor        | Lead Maintainer | Lead Maintainer  |
+|------------| -------------------- | ------------- | --------------- | ---------------- |
```
I think a space is needed between `|` and `-` such that the table renders properly. Double check the output in your local github fork.

sdake commented on manifests/profiles/PROFILES.md:
```diff
@@ -10,7 +10,7 @@ will provide further instructions. Additionally, please read the RFC requirement
+|Profile Name| K8s Platform         | Vendor        | Lead Maintainer | Lead Maintainer  |
+|------------| -------------------- | ------------- | --------------- | ---------------- |
+|openshift| OpenShift            | Red Hat / IBM | @jwendell       | @irisdingbj      |
+|| Upstream Kubernetes  | Istio         | Env WG          | discuss.istio.io |
```
extra `|` doesnt' seem to make sense to me.

brian-avery replied:
Added the extra `|` here because there's no profile name for the bottom one. After reading the comment above, perhaps it's better to say "see list at istio.io" or something?

sdake reviewed (commented):
looks good - perhaps some syntax problems with the md table.

brian-avery committed:
Remove profiles.md

istio-testing commented:
@brian-avery: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 634178f330cb2e76dd1a42721b9661a36c139f8d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/integ-pilot-k8s-tests_istio/1325837955310292992) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

brian-avery commented:
/test integ-pilot-k8s-tests_istio

istio-testing commented:
@brian-avery: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 634178f330cb2e76dd1a42721b9661a36c139f8d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28589/integ-pilot-k8s-tests_istio/1325845428612108288) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

brian-avery commented:
/test integ-pilot-k8s-tests_istio

brian-avery commented:
@sdake we moved this to documentation after talking to Frank. Should this readme be removed as well or would you like to keep it here?
